### PR TITLE
DNS templates cleanups

### PR DESCRIPTION
**Pull request by thockin**
This is a set of cleanups to the DNS templates that are used to configure the kubedns addon.  It should be reviewed commit-by-commit.

Missing from this are a handful of directories that still have old references to "skydns" names.

@errordeveloper : you are the major author of `docs/getting-started-guides/coreos/azure/addons/skydns-*`.  These files are ANCIENT forks of the DNS config files.  We can not really have this in our docs.  Can we find a way to pull from the real template files?  This is in both the main repo and teh docs repo while docs are converting.

@yifan-gu : you are the major author of `cluster/gce/coreos/kube-manifests/addons/dns/skydns-*`.  These files are forks of the DNS config files, and are drifting out of sync.  We can not really have this.  Can we find a way to pull from the real template files instead?

@mbruzek @chuckbutler : you are the major authors of `cluster/juju/layers/kubernetes/templates/skydns-*`.  These files are ANCIENT forks of the DNS config files.  We can not really have this.  Can we find a way to pull from the real template files instead?

@sdminonne : you are the major author of `cluster/libvirt-coreos/skydns-*`.  These files are ANCIENT forks of the DNS config files.  We can not really have this.  Can we find a way to pull from the real template files instead?

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29720)

### CODE REVIEW

**Comment by sdminonne** [comment]
@thockin agree: what we have today for libvirt is not reliable. Having a look to see if we can pull the real ones. Thanks

**Comment by mbruzek** [comment]
@thockin Yes we can look at ways to keep current with the skydns files. We use templating logic to pass in user defined configuration so we will have to investigate what we can do.

**Comment by girishkalele** [comment]
LGTM

File: build/kube-dns/README.md [hunk 1]
```diff
@@ -148,13 +148,16 @@ set:
 ```
 
 Second, you need to start the DNS server ReplicationController and Service. See
-the example files ([ReplicationController](../../cluster/addns/dns/skydns-rc.yaml.in) and
-[Service](../../cluster/addons/dns/skydns-svc.yaml.in)), but keep in mind that these are templated for
-Salt.  You will need to replace the `{{ <param> }}` blocks with your own values
-for the config variables mentioned above.  Other than the templating, these are
-normal kubernetes objects, and can be instantiated with `kubectl create`.
+the example files
+([ReplicationController](../../cluster/addns/dns/kubedns-rc.yaml.sed) and
```
Comment (sdminonne) [inline build/kube-dns/README.md]: @thockin typo: `../../cluster/addons/dns/...`

**Comment by thockin** [comment]
Girish said LGTM

**Comment by thockin** [comment]
rebased against the juju changes specifically for this PR :)

**Comment by errordeveloper** [comment]
@thockin for the record `docs/getting-started-guides/coreos/azure` is gone as of #23818.

**Comment by girishkalele** [comment]
@thockin I guess this was just a cleanup item, can go into the next release.

cc @MrHohn

**Comment by thockin** [comment]
I've resigned myself to needing to basically recreate this after other
changes land :)

**Comment by yifan-gu** [comment]
@thockin The plan is to use some facilities in [https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/trusty/configure-helper.sh#L864-L888](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/trusty/configure-helper.sh#L864-L888)

But I don't think I have a chance to to that until rkt's integration with CRI landed.

/cc @euank , since you are reworking the kube-up for coreos/rkt using hyperkube image, you might have a chance to touch this.

**Comment by euank** [comment]
Thanks for the ping @yifan-gu, I'm working on getting our addons to use the shared base now

**Comment by euank** [comment]
PR above, I deleted the `cluster/gce/coreos` copies of the files and pull from the templates as GCI does.

**Pull request closed without merging**